### PR TITLE
Reintroduce engine version file

### LINKED ISSUE #1840: MeiliSearch removes indexes when `data.ms` is not compatible
Steps to reproduce
- Start MeiliSearch v0.22.0
- Add documents
- Stop MeiliSearch v0.22.0 and run MeiliSearch v0.23.0 using the same `data.ms`

-> MeiliSearch starts without any error, but should not, and remove the indexes in `data.ms`

MeiliSearch should display the following error
```
Cannot open database, expected MeiliSearch engine version: X.X.X, current engine version Y.Y.Y
```
at it was before, and should not start if data.ms is not compatible without removing the indexes

### PR DESCRIPTION
**Pull request by mbStavola**
Right now if you boot up MeiliSearch and point it to a DB directory created with a previous version of MeiliSearch the existing indexes will be deleted. This [used to be](https://github.com/meilisearch/MeiliSearch/commit/51d7c84e7327b14f84522a1f5521f6d98c398a6f) prevented by a startup check which would compare the current engine version vs what was stored in the DB directory's version file, but this functionality seems to have been lost after a few refactorings of the code.

In order to go back to the old behavior we'll need to reintroduce the `VERSION` file that used to be present; I considered reusing the `metadata.json` file used in the dumps feature, but this seemed like the simpler and more approach. As the intent is just to restore functionality, the implementation is quite basic. I imagine that in the future we could build on this and do things like compatibility across major/minor versions and even migrating between formats.

Closes #1840

### CODE REVIEW

**Comment by curquiza** [comment]
Hello @mbStavola thanks for your PR
We need to discuss with the team how we want to solve this properly. We'll come back to you! 🙂

**Review by irevoire — CHANGES_REQUESTED** [review]
Hello @mbStavola, we finished debating on how we want to implement this and we decided to accept your PR.
There is quite a lot of changes. So if you realize you don’t have the time or don’t want to finish this PR please let me know and I will take care of fixing everything. 😁 

We’ll also improve our process to better follow when we break our database because currently, we don’t really know.

I also noticed that the VERSION file was not read or written when importing a [snapshot](https://docs.meilisearch.com/reference/features/snapshots.html).

**Comment by mbStavola** [comment]
@irevoire I'm definitely interested in following through and the changes suggested seem relatively minor. Please note, it might take me a day or two to get back to this, so use your best judgement/sense of urgency to determine if you'd like for me to finish this up or not.

**Comment by irevoire** [comment]
Yeeoos thank you 🎉 
I don’t really know if we are in a hurry for this PR. What do you think @curquiza?

**Comment by curquiza** [comment]
We need this bug not to be present in the next version (v0.25.0) -> the pre-release is December 6th :) According to this date, if you don't have the time @mbStavola (we would completely understand), @irevoire will need to take your branch and continue your work :)
Is it ok for you @irevoire and @mbStavola?

File: meilisearch-lib/src/index_controller/versioning/error.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#[derive(thiserror::Error, Debug)]
+pub enum VersionFileError {
+    #[error("Version file is missing")]
```
[the remark this answers is not in the document]
Comment (CaroFG) [reply meilisearch-lib/src/index_controller/versioning/error.rs:3]: "its database"

**Comment by irevoire** [comment]
Hello @mbStavola we do not have much time left before the release so I'm going to take back this PR.
I'm going to cherry-pick your commits so you'll still be listed as a contributor, thanks for your contribution! :tada:

**Comment by irevoire** [comment]
Closing this PR in favor of  #1965

**Comment by mbStavola** [comment]
@irevoire Sorry! This was a bit of a miscommunication, when I reacted to @curquiza 's comment it was to signify that I was okay handing it off since you had a deadline you wanted to meet. In retrospect a direct comment would've been better 😅

Apologies for any confusion!

**Comment by irevoire** [comment]
No problem 👍

**Pull request closed without merging**